sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -1,2 +1,66 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+
+use crate::{
+    models::BlobProperties, pipeline::StorageHeadersPolicy, BlobBlobClientGetPropertiesOptions,
+    BlobClientOptions, GeneratedBlobClient,
+};
+use azure_core::{credentials::TokenCredential, BearerTokenCredentialPolicy, Policy, Result, Url};
+use std::sync::Arc;
+
+pub struct BlobClient {
+    pub endpoint: Url,
+    pub container_name: String,
+    pub blob_name: String,
```

Comment:
Clients shouldn't have public members. We only do that for plain ol' data objects (models). We have thread-safety guarantees for clients and making them mutable violates that. You can add accessors, which is already covered in our guidelines for `endpoint`.